oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -98,6 +100,18 @@ public abstract class ExternalAccountCredentials extends GoogleCredentials {
 
   private EnvironmentProvider environmentProvider;
 
+  private static final String WORKFORCE_POOL_URL_FORMAT =
+      "https://iamcredentials.googleapis.com/v1/locations/global/workforcePools/%s/allowedLocations";
+  private static final String WORKLOAD_POOL_URL_FORMAT =
+      "https://iamcredentials.googleapis.com/v1/projects/%s/locations/global/workloadIdentityPools/%s/allowedLocations";
+
+  private static final Pattern WORKFORCE_PATTERN =
+      Pattern.compile(
+          "^//iam.googleapis.com/locations/(?<location>[^/]+)/workforcePools/(?<pool>[^/]+)/providers/(?<provider>[^/]+)$");
+  private static final Pattern WORKLOAD_PATTERN =
+      Pattern.compile(
+          "^//iam.googleapis.com/projects/(?<project>[^/]+)/locations/(?<location>[^/]+)/workloadIdentityPools/(?<pool>[^/]+)/providers/(?<provider>[^/]+)$");
```

Comment:
The regex is too strict and would only work if the universe domain is googleapis.com
I'd recommend using something like this, that just matches the part we're interested in. 
   * Workforce: r"locations/global/workforcePools/([^/]+)"
   * Workload: r"projects/([^/]+)/locations/global/workloadIdentityPools/([^/]+)"